src/components/Navigation.jsx

```diff
@@ -65,6 +143,30 @@ function Navigation() {
               </Link>
             </li>
           ))}
+          <li>
+            {userData ? (
```

Comment:
Instead of conditionally rendering this based on whether there is data back from the server, I would do this boolean based on whether there is a token in localStorage. This will prevent a situation where "Log In" shows while we are fetching data before switching to the profile image.

As a fallback, you can add a background color on the `img` so it still shows a color circle which will fill in the image when downloaded

Reply:
Done!